includes/blocks/class-sensei-course-outline-module-block.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * File containing the Sensei_Course_Outline_Module_Block class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Course_Outline_Block
+ */
+class Sensei_Course_Outline_Module_Block {
```

Comment:
Hm, we would need to pass around or create instances of them in a lot of places, (not only for the root Outline class, but pass the lesson renderer to the module too.) So since they don't have any state anyway static here seem like a good choice. Is there a reason generally to avoid it?

Reply:
The biggest one is that static methods can't be mocked so it makes classes harder to test (although there aren't any tests atm). Other than that there are the benefits of making it clearer which are the dependencies of a class and making it a bit easier to expand. Having a class with static methods methods seems more like a collection of functionality rather than an object so I think that in that case a namespace would be more fitting.

Reply:
I feel like we should solve the unit testing issue with using something better to mock things — I was checking out [AspectMock](https://github.com/Codeception/AspectMock) and it looks pretty good for this.

Having these as proper objects would work better if we were using some dependency injection framework - without that we would just move the class usage to a `new` call, but it's still not possible to mock (unless we make the property public just to test it)

Reply:
> I feel like we should solve the unit testing issue with using something better to mock things — I was checking out [AspectMock](https://github.com/Codeception/AspectMock) and it looks pretty good for this.

Not sure how well this library is maintained. Last release was on February and previous one was a year ago.

> Having these as proper objects would work better if we were using some dependency injection framework - without that we would just move the class usage to a `new` call, but it's still not possible to mock (unless we make the property public just to test it)

No need to change visibility, you can always use [reflection](https://www.php.net/manual/en/reflectionproperty.setaccessible.php) to make it accesible.

Reply:
Went through this in 398fd71 and moved everything to instance members, adding `Sensei_Blocks` as `Sensei()->blocks`. This should help with testing/mocking while also keeping everything accessible.